File: 1295_Find_Number_With_Even_Number_of_Digits/Solution.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <math.h>
using namespace std;

class Solution
{
public:
    int findNumbers(vector<int> &nums)
    {
        int count = 0;
        for (auto const num : nums)
        {
            if (((int(log10(num) + 1)) % 2) == 0) // use logarithmic finction to get number of digits and then do modulo operation
            {
                count++;
            }
        }
        return count;
    }
};

File: 1295_Find_Number_With_Even_Number_of_Digits/Solution.java
class Solution {
    public int findNumbers(int[] nums) {
        int count = 0;
        for (int n : nums) {
            if (String.valueOf(n).length() % 2 == 0) { // this is to parse int to string to use the length ffunction
                count += 1;
            }
        }

        return count;

    }
}